lib/mayaUsd/fileio/primReaderContext.h

```diff
@@ -90,6 +102,9 @@ class UsdMayaPrimReaderContext
     // used to keep track of prims that are created.
     // for undo/redo
     ObjectRegistry* _pathNodeMap;
+
+    // Tracks new nodes:
+    std::shared_ptr<ObjectRegistry> _trackedNewMayaNodes;
```

Comment:
Can we switch to a small vector of objects instead of a map? We know that this container will not have many elements and would be great to leverage this information.